validator/src/main/java/com/google/daq/mqtt/validator/Validator.java

```diff
@@ -646,10 +654,33 @@ private void handleUdmiConfig(UdmiConfig udmiConfig) {
     JsonUtil.writeFile(udmiConfig, new File(outBaseDir, UDMI_CONFIG_JSON_FILE));
   }
 
+  private boolean handleMetadataUpdate(Map<String, String> attributes, Object messageObject) {
+    String deviceId = attributes.get("deviceId");
+    String subFolderRaw = attributes.get(SUBFOLDER_PROPERTY_KEY);
+    String subTypeRaw = attributes.get(SUBTYPE_PROPERTY_KEY);
+
+    if (SubType.MODEL.value().equals(subTypeRaw) && SubFolder.UPDATE.value().equals(subFolderRaw)) {
+      if (reportingDevices.containsKey(deviceId)) {
+        ReportingDevice device = reportingDevices.get(deviceId);
+        Metadata metadata = convertTo(Metadata.class,
+            requireNonNull(messageObject, "messageObject is null"));
+
+        if (metadata.cloud != null && metadata.cloud.operation == Operation.DELETE) {
```

Comment:
observation: you can also do (I think):
```
if (catchToNull(() -> metadata.cloud.operation) == Operation.DELETE) {
  ...
}
```

Functionally they are the same, but just written different -- purely a matter of style.  Totally up to you which you like better (I tend to over abstract things sometimes).